src/gsi/hybrid_ensemble_isotropic.F90

```diff
@@ -5596,31 +5596,25 @@ subroutine setup_ensgrp2aensgrp
      enddo
   enddo
 
-  if (i_ensloccov4tim==0) then
-     ensloccov4tim=one
-  elseif (i_ensloccov4tim==1)then
-     ensloccov4tim=zero
+  if (r_ensloccov4tim>=zero.and.r_ensloccov4tim<=one) then
+     ensloccov4tim=r_ensloccov4tim
      ensloccov4tim(1)=one
   else
-     write(6,*)'setup_ensgrp2aensgrp: wrong i_ensloccov4tim'
+     write(6,*)'setup_ensgrp2aensgrp: wrong r_ensloccov4tim'
      call stop2(666)
   endif
-  if (i_ensloccov4var==0) then
-     ensloccov4var=one
-  elseif (i_ensloccov4var==1)then
-     ensloccov4var=zero
+  if (r_ensloccov4var>=zero.and.r_ensloccov4var<=one) then
```

Comment:
The equality issue also exist for those ">= " and " <=".   what will be the results when r_ensloccov4var = zero or one?